presto-main/src/main/java/com/facebook/presto/operator/scalar/GroupingOperationFunction.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.operator.Description;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.SqlType;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.List;
+
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.StandardTypes.BIGINT;
+
+public final class GroupingOperationFunction
+{
+    private GroupingOperationFunction() {}
+
+    @ScalarFunction
+    @Description("Returns a bitmap as an integer, indicating which columns are present in the grouping")
+    @SqlType(BIGINT)
+    public static long grouping(
+            @SqlType(BIGINT) long groupId,
+            @SqlType("array(integer)") Block groupingOrdinalsBlock,
+            @SqlType("array(array(integer))") Block groupingSetOrdinalsBlock)
+    {
+        List<Integer> groupingOrdinals = new ArrayList<>(groupingOrdinalsBlock.getPositionCount());
```

Comment:
The only way to do that today is to create a synthetic type like we did with regexp patterns, introduce a functions that can transform the grouping ordinals and grouping sets arrays into such an object, and let the constant folder turn them into an object of that type before code generation.